### PR TITLE
Prerelease adjustments

### PR DESCRIPTION
**Pull request by mauricioirace**
Changes:
- Improved README.MD based on lumino's
- Fixed key paths in lumino.json5
- Removed package-lock.json
- Removed old configuration files

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,97 +10,106 @@
 ![](https://img.shields.io/badge/npm-%3E%3D6.0.0-orange.svg?style=flat-square)
 ![](https://img.shields.io/badge/Node.js-%3E%3D10.0.0-orange.svg?style=flat-square)
 
-> Simple boot node for RIF Communications PubSub
+# RIF Communications bootnode
 
-The aim to provide a libp2p node which serves as bootstrap node for projects which use [`rif-commmunications-pubsub`](https://github.com/rsksmart/rif-communications-pubsub). 
+## Getting started:
```
Comment (mortelli) [inline README.md]:
```suggestion
## Getting started
```

(no need for `:` in titles)

File: README.md [hunk 2]
```diff
@@ -10,97 +10,106 @@
...
-## Table of Contents
+### Get the code
 
-- [RIF Communications PubSub node](#rif-communications-pubsub-node)
-  - [Table of Contents](#table-of-contents)
-  - [Usage](#usage)
-  - [Config](#config)
-  - [Docker](#docker)
-  - [License](#license)
+Download the code by executing `git clone https://github.com/rsksmart/rif-communications-pubsub-bootnode -b grpc-api`.
```
Comment (mortelli) [inline README.md]: i would use ``` for this command and enter it in a new line

File: README.md [hunk 3]
```diff
@@ -10,97 +10,106 @@
...
 ![](https://img.shields.io/badge/Node.js-%3E%3D10.0.0-orange.svg?style=flat-square)
 
-> Simple boot node for RIF Communications PubSub
+# RIF Communications bootnode
 
-The aim to provide a libp2p node which serves as bootstrap node for projects which use [`rif-commmunications-pubsub`](https://github.com/rsksmart/rif-communications-pubsub). 
+## Getting started:
 
-This can be also used for local development where you can define the list of Rooms that will be listened on and messages printed out to STDOUT.
+- Node 14+ (14.13 recommended)
```
Comment (mortelli) [inline README.md:19]:
```suggestion
### Prerequisites
- Node 14+ (14.13 recommended)
```

File: README.md [hunk 4]
```diff
@@ -10,97 +10,106 @@
...
 
-Example of usage:
 
-```
-npm start
-```
+### Create a key for the RIF Communications bootnode
 
-Spawns a new libp2p node with new PeerId listening to TCP connections on port 6030.
+1. Create the `config/keys/lumino` folder.
```
Comment (mortelli) [inline README.md]:
```suggestion
1. Create the `config/keys/client` folder inside the cloned directory.
```

File: README.md [hunk 5]
```diff
@@ -10,97 +10,106 @@
...
+2. Execute the following commands inside this folder:
```
Comment (mortelli) [inline README.md]:
```suggestion
2. Execute the following commands inside this new folder:
```

File: README.md [hunk 6]
```diff
@@ -10,97 +10,106 @@
...
-<roomName>: message {from: <peerId>, data: <content of the message>, to: <peerId>} // Only for direct messages
+openssl pkcs8 -in ec_key.pem -topk8 -v2 aes-256-cbc -v2prf hmacWithSHA256 -outform DER -out ec_key_pkcs8_v2.der
+``` 
+The latter one will require you to define a password.
+
+### Edit the private key configuration
+
+- go back to `config` folder
+- edit the `lumino.json5` file. Modify the `key.password` value to your key's password
+- change the `key.privateKeyURLPath` to the config key path `[...]/config/keys/lumino/ec_key_pkcs8_v2.der`
```
Comment (mortelli) [inline README.md]:
```suggestion
1. Go back to the `config` folder
2. Edit the `client.json5` file:
    1. Modify the `key.password` value to your key's password
    2. Change the `key.privateKeyURLPath` to the config key path `[...]/config/keys/client/ec_key_pkcs8_v2.der`
```

File: README.md [hunk 7]
```diff
@@ -10,97 +10,106 @@
...
+
+
+### Configure the Bootstrap nodes
+
+In order to access the network, you need to connect to at least one node
```
Comment (mortelli) [inline README.md]:
```suggestion
In order to access a network, you need to connect to at least one node 
```

File: README.md [hunk 8]
```diff
@@ -10,97 +10,106 @@
...
+
+- go back to `config` folder
+- edit the `lumino.json5` file. Modify the `key.password` value to your key's password
+- change the `key.privateKeyURLPath` to the config key path `[...]/config/keys/lumino/ec_key_pkcs8_v2.der`
+
+
+### Configure the Bootstrap nodes
+
+In order to access the network, you need to connect to at least one node 
+that's are already in it. You can configure them using the the following parameters:
```
Comment (mortelli) [inline README.md]:
```suggestion
that's are already in it. You can configure these using the the following parameters:
```

File: README.md [hunk 9]
```diff
@@ -10,97 +10,106 @@
...
+```json5
+{
+  libp2p: {
+    config: {
+      peerDiscovery: {
+        bootstrap: {
+          // Enable bootstraping
+          enabled: true,
+          // list of nodes to connect by default
+          // e.g. Lumino Networkd bootstrap nodes
```
Comment (mortelli) [inline README.md]:
```suggestion
          // e.g. Lumino Network bootstrap nodes
```

File: README.md [hunk 10]
```diff
@@ -10,97 +10,106 @@
...
+In order to access the network, you need to connect to at least one node 
+that's are already in it. You can configure them using the the following parameters:
+
+```json5
+{
+  libp2p: {
+    config: {
+      peerDiscovery: {
+        bootstrap: {
+          // Enable bootstraping
```
Comment (mortelli) [inline README.md]:
```suggestion
          // Enable bootstrapping
```

File: README.md [hunk 11]
```diff
@@ -10,97 +10,106 @@
...
+      }
+    }
+  }
+}
 ```
+### Start the node
 
-## Config
+- At root folder, run `NODE_ENV=lumino npm run api-server`
+- You should see a log output similar to this one:
```
Comment (mortelli) [inline README.md]:
```suggestion
At the project root folder, run `NODE_ENV=client npm run api-server`

You should see a log output similar to this one:
```

File: README.md [hunk 12]
```diff
@@ -10,97 +10,106 @@
...
-<roomName>: message {from: <peerId>, data: <content of the message>}
-<roomName>: message {from: <peerId>, data: <content of the message>, to: <peerId>} // Only for direct messages
+openssl pkcs8 -in ec_key.pem -topk8 -v2 aes-256-cbc -v2prf hmacWithSHA256 -outform DER -out ec_key_pkcs8_v2.der
+``` 
+The latter one will require you to define a password.
+
+### Edit the private key configuration
+
+- go back to `config` folder
+- edit the `lumino.json5` file. Modify the `key.password` value to your key's password
```
Comment (mortelli) [inline README.md]: despite PR #33, i think we should probably change this. we shouldn't need to make a direct reference to `lumino` in these configs.

i propose we keep the `lumino.json5` around, but also add another base config (`client.json5`) without the bootnotes, and then make a reference to it in these sections.

File: README.md [hunk 13]
```diff
@@ -10,97 +10,106 @@
...
-
-You can generate one using `npm run generate-peerid` and then set that either with config file or `RIFC_PEER_ID` env. variable (set the variable as the whole generated JSON).
-
-```
-$ PEER_ID=$(npm run generate-peerid) // This should be stored in some file somewhere
-$ docker build -t rif-comunication-bootnode .  
-$ docker run -e RIFC_PEER_ID="$PEER_ID" -p 6666 -p 6667 -it rif-comunication-bootnode  
+## Additional settings
+
+### Exposing RIF Communications from a private IP address
```
Comment (marcosmartinez7) [inline README.md:100]: 👏🏻

**Review by mortelli — APPROVED** [review]

**Review by marcosmartinez7 — APPROVED** [review]
Great improvement. 

I think that we should save the production.json5 file that we used for the release in some way. Maybe it must be on the Lumino repo. What do you think?

**Review by JONAF2103 — APPROVED** [review]
LGTM

File: README.md (hunk 12 above)
Comment (mauricioirace) [reply README.md]: Agree. I'd keep lumino.json5 but also add client.json5. I'll change all the references to lumino.json5 to client.json5. Agree?

**Comment by mauricioirace** [comment]
> Lumino

That's prod.json5
I think we could add it, but as you say, I'd rather add it to lumino's repo somewhere, as it's lumino specific configuration, and there are actually 11 different configurations (8 + 3)

**Pull request merged**